FrontEnd/src/components/ProfileDetail/DetailedInfo/PhoneEmail.jsx

```diff
@@ -0,0 +1,104 @@
+import { useState } from 'react';
+import useSWR from 'swr';
+import { PropTypes } from 'prop-types';
+import classes from './PhoneEmail.module.css';
+
+function PhoneEmail ({ isAuthorized, profileId }) {
+    const [isPhoneShown, setPhoneShown] = useState(false);
+    const [isEmailShown, setEmailShown] = useState(false);
+    const authToken = localStorage.getItem('Token');
+
+    // TODO: change the logic of getting user.id when PR with hooks will be merged
+
+    const { data: userData } = useSWR(
+        authToken ? `${process.env.REACT_APP_BASE_API_URL}/api/auth/users/me/` : null,
+        url =>
+            fetch(url, {
+                method: 'GET',
+                headers: {
+                    'Authorization': `Token ${authToken}`,
+                },
+            }).then(res => res.json()),
+    );
+
+    const { data: profileData} = useSWR(
+        authToken ? `${process.env.REACT_APP_BASE_API_URL}/api/profiles/${profileId}?with_contacts=True` : null,
+        url =>
+            fetch(url, {
+                method: 'GET',
+                headers: {
+                    'Authorization': `Token ${authToken}`,
+                },
+            }).then(res => res.json()),
+    );
+
+    const urlViewed = `${process.env.REACT_APP_BASE_API_URL}/api/viewed-list/`;
+    const viewedData = userData && {user: userData.id, company: profileId,};
+
+    async function sendRequest(url, data ) {
+        return fetch(url, {
+          method: 'POST',
+          headers: {
+            Authorization: `Token ${authToken}`,
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify(data),
+        }).then();
+    }
+
+    const handlePhoneClick = async () => {
+        setPhoneShown(true);
+        try {
+            await sendRequest(
+                urlViewed,
+                viewedData
+            );
+        } catch (error) {
+            console.error(error);
```

Comment:
Toast message for errors

Reply:
Should we notify user about unsuccessful attempt to add info about his action in db, when he try to see hidden phone or email?

Reply:
Do we use this endpoint for analytics? If yes, probably error message is redundant